pkg/clustermanager/cluster_manager.go

```diff
@@ -727,7 +728,7 @@ func (c *ClusterManager) waitForMachineDeploymentReplicasReady(ctx context.Conte
 		timeout = c.machinesMinWait
 	}
 
-	r := retrier.New(timeout)
+	r := retrier.NewWithProportionalBackOff(timeout, waitProportion)
```

Comment:
This can lead to very long wait times if there is a large number of machines.
Imagine 20 machines. Total wait time will be 200 minutes. So the backoff time will be 2 minutes.

What do you think about implementing something similar to what we have for `waitForNodesReady`?
The backoff time is dynamic, so the more machines unready, the more we wait in between retries. They more machines ready, the less we wait. This same file, line 740